### PR TITLE
Fixing Zebra Async Timeouts

### PR DESCRIPTION
**Pull request by gerardtoconnor**
On my original PR there was a bug in the async binding that was causing requests to timeout and perform poorly, this fixes that with different state-machine design.

I have also added 'fortunes' test as well as simplified server variation

### CODE REVIEW

**Comment by gerardtoconnor** [comment]
Fortunes test is failing build (and locking server)

Working off other asp.net core builds & Giraffe I used the postgres connection string
`"Server=tfb-database;Database=hello_world;User Id=benchmarkdbuser;Password=benchmarkdbpass;Maximum Pool Size=1024;NoResetOnClose=true;Enlist=false;Max Auto Prepare=3"`
Assuming there is a local shared postgres instance that is used, is this not the case?

When I run with a connection string to my local postgres server it works fine

**Comment by gerardtoconnor** [comment]
@dv00d00 can you assist? replicated the postgres connection string you used presuming there is a shared local server but that doesn't work and fails build, I didn't see where you or anyone else creates postgres server instance?

**Comment by nbrady-techempower** [comment]
@gerardtoconnor `database:` needs to be `"Postgres"`. This is how the suite knows which db to start for your test.
 https://github.com/TechEmpower/FrameworkBenchmarks/blob/0162d4c96f808032ab7c690c13cdd89aca7b720e/frameworks/FSharp/Zebra/benchmark_config.json#L12

**Comment by gerardtoconnor** [comment]
@nbrady-techempower Thanks so much, that was the issue, have just pushed update there, greatly appreciated.

**Comment by gerardtoconnor** [comment]
The build on FSharp is failing with 
```
Running Test: suave
--------------------------------------------------------------------------------
[0m[2msuave: [0m[37m[1mStep 1/7 : FROM microsoft/dotnet:2.1-sdk-stretch[0m


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

Don't know if it's an issue with docker image retrieval or what but this was tests already passing that I have not touched!?

**Comment by msmith-techempower** [comment]
I will reset the test.

**Comment by gerardtoconnor** [comment]
Awesome, thanks @msmith-techempower, hopefully the image fetch will not timeout this time.

**Comment by gerardtoconnor** [comment]
Dockerhub failed again 😖. I've only touched 
`FSharp\Zebra` so if you can just test that, would be good for merge?

**Comment by msmith-techempower** [comment]
Yeah, I do not know what's going on with Travis/Docker; I'll pull your branch locally and test.

File: frameworks/FSharp/Zebra/src/App/Program.fs [hunk 1]
```diff
@@ -4,35 +4,132 @@ open State
...
+                    ]
+                    bindFor<_,_> (fun ls -> ls :> seq<Fortune> ) (
+                        tr [] [
+                            td [] [ bindInt (fun v -> v.id) ]
+                            td [] [ bindStr (fun v -> v.message) ]
+                        ]
+                    )
+                ]
+            ]
+        ] |> compileDoc
```
Comment (msmith-techempower) [inline frameworks/FSharp/Zebra/src/App/Program.fs]: Just out of curiosity, what is happening here? I recall older implementations using a Razor template for `fortunes`, but this seems to be a simpler compiled template... thing.

File: frameworks/FSharp/Zebra/src/App/Program.fs [hunk 2]
```diff
@@ -4,35 +4,132 @@ open State
...
+                    xs.Add View.extra
+                    xs.Sort View.FortuneComparer
+                    xs
+
+                ctx.Render( fortunes, View.fortuneView )
+                }
         ]
 
+    // Simple implimentation
+    let plaintextPrint = Simple.textFn "Hello World!"
```
Comment (msmith-techempower) [inline frameworks/FSharp/Zebra/src/App/Program.fs]: Missing a required comma.

File: frameworks/FSharp/Zebra/src/App/Program.fs (hunk 1 above)
Comment (gerardtoconnor) [reply frameworks/FSharp/Zebra/src/App/Program.fs]: "Thing" is probably the best word to describe it alright … F# can use Razor but it's bit of a hack together of XML & C#, in F# we can us a simple DSL to represent the HTML as well as type checking all the bindings. `body`, `table` etc are all just functions that create a DSL node tree similar to XML but with the `compileDoc` it is taking in all the DSL elements and creating a processing array of either the static elements (grouped into one byte array), or a binding function that will inject values at runtime.

File: frameworks/FSharp/Zebra/src/App/Program.fs (hunk 2 above)
Comment (gerardtoconnor) [reply frameworks/FSharp/Zebra/src/App/Program.fs]: Ahhh, sorry, such a rookie mistake

**Comment by gerardtoconnor** [comment]
Yay, finally, Dockerhub working as well as Zebra (and it's missing comma)

```
Verification Summary

--------------------------------------------------------------------------------

| suave
|       plaintext     : PASS
|       json          : PASS

| zebra
|       plaintext     : PASS
|       json          : PASS
|       fortune       : PASS

| zebra-simple
|       plaintext     : PASS
|       json          : PASS

================================================================================ 
```

File: frameworks/FSharp/Zebra/src/App/Program.fs (hunk 1 above)
Comment (msmith-techempower) [reply frameworks/FSharp/Zebra/src/App/Program.fs]: That's neat.

**Comment by gerardtoconnor** [comment]
Thanks for all your assistance @msmith-techempower 👍

**Pull request merged**